### PR TITLE
Add address search & fields to reg form

### PR DESCRIPTION
**Pull request by rowanhogan**

### CODE REVIEW

File: source/components/AddressSearch/index.js [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+import TextField from '../TextField'
+
+class AddressSearch extends Component {
+  constructor () {
+    super()
+    this.handleChange = this.handleChange.bind(this)
+    this.handleKeyDown = this.handleKeyDown.bind(this)
+    this.setSelected = this.setSelected.bind(this)
+    this.state = {
+      q: '',
```
Comment (kizza) [inline source/components/AddressSearch/index.js]: `q` equals query?  Single letter variables make me think :)

Comment (rowanhogan) [reply source/components/AddressSearch/index.js]: Indeed it does... This PR is _mostly_ a copy/paste job from MyMarathon, hence its speedy release despite largish amount of code. So blame past Rowan for that one!

Comment (kizza) [reply source/components/AddressSearch/index.js]: lol! I get it :)  _That past rowan guy..._  No worries

Comment (rowanhogan) [reply source/components/AddressSearch/index.js]: It's a fair criticism though, I will update!

**Pull request merged**